enhancements/cli_manager.md

```diff
@@ -0,0 +1,166 @@
+Some current examples are oc, kubectl, odo, istio, tekton, and knative.  It is difficult for users to discover what tools exist, 
+where to download them from and which version they should download.  We need to simplify as much as possible the interaction
+of services on OpenShift.  We need a mechanism for providing and consuming tools that is simple to add on to as new CLIs are 
+developed from a variety of sources - and this needs to be managed from within a cluster.
+This is the motivation for the CLI manager.
+
+### Goals
+
+Each component wishing to provide customers with their binaries will build and publish them via an official channel to an
+index location from which the CLI manager will obtain the files.
+
+Possible routes for supplying CLI binaries:    
+1.  Central index that has links to where artifacts are stored
+2.  Central repository where all artifacts are stored - [index of /pub/openshift-v4/clients](https://mirror.openshift.com/pub/openshift-v4/clients/)
+currently is where crc, oc, ocp-dev-preview, ocp, odo, and serverless artifacts are published.  
+3.  Images for each CLI. The CLI-manager will provide an extract mechanism, similar to how we currently 
+`oc adm release extract --command oc` and `oc image extract` for each CLI image, and a CR for each CLI will provide the CLI-manager information about each CLI image.
+
+Regardless of the index scenario above, CLI-manager will be given the location of each CLI's artifacts via the CLI's CR.    
+
+Option 2 plus Option 3 is the option we as a team are leaning toward.  It requires that each CLI publish artifacts to [index of /pub/openshift-v4/clients](https://mirror.openshift.com/pub/openshift-v4/clients/)
+then create an image similar to these: [example images for oc cli-manager](https://github.com/sallyom/clustercli/tree/master/hack/cli-images)
+The CLI manager will gather information from each CLI's CR about each CLI image.  In disconnected environments this will be through a mirrored local 
+registry.  Extracting binaries from a CLI image will be similar to how we currently extract the installer and oc from a release payload.
+`oc cli-manager` will enable installing CLIs with known-compatible versions for a cluster.  CLIs will be managed through extending the oc commands we currently
+have that extract `oc` and `openshift-install` binaries from the release payload.  The logic for `oc image extract` will be re-used and extended for
+extracting all available openshift CLIs. As an example, the following command is possible today:
+
+```console
+$ oc image extract quay.io/sallyom/kn:latest --path /tmp/amd64/kn.tar.gz:$HOME/openshift-clis/compressed/
+$ tar xvf $HOME/openshift-clis/compressed/kn.tar.gz -C $HOME/openshift-clis/
+```
+
+As laid out in this proposal, each CLI will provide a separate image.
+Each CLI is responsible for creating a CR to hold metadata that the CLI-manager will need.
```

Comment:
creating a metadata/description of deliverable binary.

Reply:
updated